tests/components/webostv/test_media_player.py

```diff
@@ -475,6 +475,28 @@ async def test_client_disconnected(
     assert "TimeoutError" not in caplog.text
 
 
+async def test_client_key_update_on_connect(
+    hass: HomeAssistant,
+    client,
+    monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test client key update upon connect."""
+    config_entry = await setup_webostv(hass)
+
+    assert config_entry.data[CONF_CLIENT_SECRET] == client.client_key
+
+    monkeypatch.setattr(client, "is_connected", Mock(return_value=False))
+    monkeypatch.setattr(client, "client_key", "new_key")
```

Comment:
```suggestion
    client.is_connected.return_value = False
    client.client_key = "new_key"
```

Reply:
I think your approach is correct, if it is not needed I prefer not to use it, I am a little bit cautious about modifying all tests to remove it, but will slowly do it if it doesn't break anything